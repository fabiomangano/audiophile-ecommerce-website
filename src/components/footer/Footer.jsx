import React from 'react';
import { Link } from 'react-router-dom';
import { useMediaQuery } from 'react-responsive';
import { ReactComponent as Logo } from '../../assets/shared/desktop/logo.svg';
import { MediumOrLargeScreen, LargeScreen } from '../../mediaQueries';
import NavLinks from '../nav-links/NavLinks';
import SocialLinks from '../social-links/SocialLinks';
import Marginer from '../marginer/Marginer';
import style from './Footer.module.scss';

const Footer = () => {
  const isMediumOrLargeScreen = useMediaQuery(MediumOrLargeScreen);
  const isLargeScreen = useMediaQuery(LargeScreen);
  const navLinksDirection = isMediumOrLargeScreen ? 'horizontal' : 'vertical';

  return (
    <footer>
      <div className={style.row} />
      <div className={style.container}>
        <div className={style.logo_navlinks}>
          <Link to="/">
            <Logo />
          </Link>

          {!isLargeScreen ? (
            <Marginer direction="horizontal" margin={3} />
          ) : null}

          <NavLinks direction={navLinksDirection} />
        </div>

        <Marginer direction="horizontal" margin={3} />

        <p className={style.about}>
          Audiophile is an all in one stop to fulfill your audio needs.
          We&#39;re a small team of music lovers and sound specialists who are
          devoted to helping you get the most out of personal audio. Come and
          visit our demo facility - weâ€™re open 7 days a week.
        </p>

        <Marginer direction="horizontal" margin={3} />

        <div className={style.social_links_copyright}>
          <SocialLinks />
          {!isMediumOrLargeScreen ? (
            <Marginer direction="horizontal" margin={3} />
          ) : null}
          <p className={style.copyright}>Copyright 2021. All Rights Reserved</p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
