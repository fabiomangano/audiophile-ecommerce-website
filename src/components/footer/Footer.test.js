import React from 'react';
import ShallowRender from 'react-test-renderer/shallow';
import { useMediaQuery } from 'react-responsive';
import Footer from './Footer';

jest.mock('react-responsive', () => {
  const originalModule = jest.requireActual('react-responsive');
  return {
    __esModule: true,
    ...originalModule,
    useMediaQuery: jest.fn(),
  };
});

describe('Footer', () => {
  beforeEach(() => {
    useMediaQuery.mockClear();
  });

  afterAll(() => {
    useMediaQuery.mockRestore();
  });

  it('should render correctly when vw is large or more', () => {
    useMediaQuery.mockReturnValueOnce(false).mockReturnValueOnce(true);
    const renderer = new ShallowRender();
    renderer.render(<Footer />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('should render correctly when vw is medium more', () => {
    useMediaQuery.mockReturnValueOnce(true).mockReturnValueOnce(false);
    const renderer = new ShallowRender();
    renderer.render(<Footer />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('should render correctly when vw is small or less', () => {
    useMediaQuery.mockReturnValueOnce(false).mockReturnValueOnce(false);
    const renderer = new ShallowRender();
    renderer.render(<Footer />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
