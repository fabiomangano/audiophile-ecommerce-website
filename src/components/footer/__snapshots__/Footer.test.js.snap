// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly when vw is large or more 1`] = `
<footer>
  <div
    className="row"
  />
  <div
    className="container"
  >
    <div
      className="logo_navlinks"
    >
      <Link
        to="/"
      >
        <ForwardRef(SvgLogo) />
      </Link>
      <NavLinks
        direction="vertical"
      />
    </div>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <p
      className="about"
    >
      Audiophile is an all in one stop to fulfill your audio needs. We're a small team of music lovers and sound specialists who are devoted to helping you get the most out of personal audio. Come and visit our demo facility - we’re open 7 days a week.
    </p>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <div
      className="social_links_copyright"
    >
      <SocialLinks />
      <Marginer
        direction="horizontal"
        margin={3}
      />
      <p
        className="copyright"
      >
        Copyright 2021. All Rights Reserved
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer should render correctly when vw is medium more 1`] = `
<footer>
  <div
    className="row"
  />
  <div
    className="container"
  >
    <div
      className="logo_navlinks"
    >
      <Link
        to="/"
      >
        <ForwardRef(SvgLogo) />
      </Link>
      <Marginer
        direction="horizontal"
        margin={3}
      />
      <NavLinks
        direction="horizontal"
      />
    </div>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <p
      className="about"
    >
      Audiophile is an all in one stop to fulfill your audio needs. We're a small team of music lovers and sound specialists who are devoted to helping you get the most out of personal audio. Come and visit our demo facility - we’re open 7 days a week.
    </p>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <div
      className="social_links_copyright"
    >
      <SocialLinks />
      <p
        className="copyright"
      >
        Copyright 2021. All Rights Reserved
      </p>
    </div>
  </div>
</footer>
`;

exports[`Footer should render correctly when vw is small or less 1`] = `
<footer>
  <div
    className="row"
  />
  <div
    className="container"
  >
    <div
      className="logo_navlinks"
    >
      <Link
        to="/"
      >
        <ForwardRef(SvgLogo) />
      </Link>
      <Marginer
        direction="horizontal"
        margin={3}
      />
      <NavLinks
        direction="vertical"
      />
    </div>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <p
      className="about"
    >
      Audiophile is an all in one stop to fulfill your audio needs. We're a small team of music lovers and sound specialists who are devoted to helping you get the most out of personal audio. Come and visit our demo facility - we’re open 7 days a week.
    </p>
    <Marginer
      direction="horizontal"
      margin={3}
    />
    <div
      className="social_links_copyright"
    >
      <SocialLinks />
      <Marginer
        direction="horizontal"
        margin={3}
      />
      <p
        className="copyright"
      >
        Copyright 2021. All Rights Reserved
      </p>
    </div>
  </div>
</footer>
`;
